recipes/quickstart/Multi-Modal-RAG/README.md

```diff
@@ -0,0 +1,116 @@
+# End to End Tutorial on using Llama models for Multi-Modal RAG 
+
+## Recipe Overview: Multi-Modal RAG using `Llama-3.2-11B` model: 
+
+This is a complete workshop on labelling images using the new Llama 3.2-Vision Models and performing RAG using the image caption capiblites of the model.
+
+- **Data Labeling and Preparation:** We start by downloading 5000 images of clothing items and labeling them using `Llama-3.2-11B-Vision-Instruct` model
+- **Cleaning Labels:** With the labels based on the notebook above, we will then clean the dataset and prepare it for RAG
+- **Building Vector DB and RAG Pipeline:** With the final clean dataset, we can use descriptions and 11B model to generate recommendations
+
+## Requirements:
+
+Before we start:
+
+1. Please grab your HF CLI Token from [here](https://huggingface.co/settings/tokens)
+2. git clone [this dataset](https://huggingface.co/datasets/Sanyam/MM-Demo) inside the Multi-Modal-RAG folder: `git clone https://huggingface.co/datasets/Sanyam/MM-Demo`
+3. Launch jupyter notebook inside this folder
+4. We will also run two scripts after the notebooks
+5. Make sure you grab a together.ai token [here](https://www.together.ai)
+
+## Detailed Outline for running:
+
+Order of running files, the notebook establish the method of approaching the problem. Once we establish it, we use the scripts to run the method end to end.
+
+- Notebook 1: `Part_1_Data_Preperation.ipynb`
+- Script: `label_script.py`
+- Notebook 2: `Part_2_Cleaning_Data_and_DB.ipynb`
+- Notebook 3: `Part_3_RAG_Setup_and_Validation.ipynb`
+- Script: `final_demo.py`
+
+Here's the detailed outline:
+
+### Step 1: Data Prep and Synthetic Labeling:
+
+[Notebook for Step 1](./notebooks/Part_1_Data_Preperation.ipynb) and [Script for Step 1](./scripts/label_script.py)
+
+To run the script (remember to set n):
+```
+python scripts/label_script.py --hf_token "your_huggingface_token_here" \
+    --input_path "../MM-Demo/images_compressed" \
+    --output_path "../MM-Demo/output/" \
+    --num_gpus N
+```
+
+The dataset consists of 5000 images with some meta-data.
+
+The first half is preparing the dataset for labeling:
+- Clean/Remove corrupt images
+- EDA to understand existing distribution
+- Merging up categories of clothes to reduce complexity 
+- Balancing dataset by randomly sampling images
```

Comment:
why balancing matter here as we are not FT the model? maybe add a line to explain.